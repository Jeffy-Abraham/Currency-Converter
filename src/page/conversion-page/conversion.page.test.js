import React from "react";
import Conversion from "./conversion.page";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure, mount, render } from "enzyme";
import ListStateProvider from "../../context/listExchangeRate-context/listExchangeRate-context";

import renderer from "react-test-renderer";

configure({ adapter: new Adapter() });
const setUp = (props = {}) => {
  const component = shallow(
    <ListStateProvider
      value={{ AED: 3.67295, AFN: 77.199999, ALL: 102.35, AMD: 522.721686 }}
    >
      <Conversion />
    </ListStateProvider>
  ).dive();
  console.log(component.debug());
  return component;
};
describe("A snapshot is created for ConversionPage", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <ListStateProvider
          value={{ AED: 3.67295, AFN: 77.199999, ALL: 102.35, AMD: 522.721686 }}
        >
          <Conversion />
        </ListStateProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

  