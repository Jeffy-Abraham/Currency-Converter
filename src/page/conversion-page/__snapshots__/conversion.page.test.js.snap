// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A snapshot is created for ConversionPage renders correctly 1`] = `
<div
  className="Grid-Layout"
>
  <div
    className="Grid-Heading"
  >
    <div
      className="Grid-Title"
    >
      <span>
        US Dollar (USD) Exchange Rates
      </span>
    </div>
  </div>
  <div
    className="Display-Table-Container"
  >
    <table
      className="Display-Table"
    >
      <tr
        className="Display-Table-Rows"
      >
        <th>
          Currency
        </th>
        <th>
          Currency Name
        </th>
        <th>
          Exchange Rate= 1USD
        </th>
      </tr>
      <span
        className="css-1rpz77w"
      />
    </table>
  </div>
</div>
`;
